pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,644 @@
+        >>> new_batch, post_processing_fn = program(qscripts)
+        >>> results = dev.execute(new_batch, execution_config=execution_config)
+        >>> post_processing_fn(results)
+
+        If you monitor your CPU usage, you should see 5 new Python processes pop up to
+        crunch through those ``QuantumScript``'s. Beware not oversubscribing your machine.
+        This may happen if a single device already uses many cores, if NumPy uses a multi-
+        threaded BLAS library like MKL or OpenBLAS for example. The number of threads per
+        process times the number of processes should not exceed the number of cores on your
+        machine. You can control the number of threads per process with the environment
+        variables:
+
+        * ``OMP_NUM_THREADS``
+        * ``MKL_NUM_THREADS``
+        * ``OPENBLAS_NUM_THREADS``
+
+        where the last two are specific to the MKL and OpenBLAS libraries specifically.
+
+        .. warning::
+
+            Multiprocessing may fail depending on your platform and environment (Python shell,
+            script with a protected entry point, Jupyter notebook, etc.) This may be solved
+            changing the so-called start method. The supported start methods are the following:
+
+            * Windows (win32): spawn (default).
+            * macOS (darwin): spawn (default), fork, forkserver.
+            * Linux (unix): spawn, fork (default), forkserver.
+
+            which can be changed with ``multiprocessing.set_start_method()``. For example,
+            if multiprocessing fails on macOS in your Jupyter notebook environment, try
+            restarting the session and adding the following at the beginning of the file:
+
+            .. code-block:: python
+
+                import multiprocessing
+                multiprocessing.set_start_method("fork")
+
+            Additional information can be found in the
+            `multiprocessing doc <https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods>`_.
+
+
+    """
+
+    @property
+    def name(self):
+        """The name of the device."""
+        return "default.clifford"
+
+    # pylint:disable = too-many-arguments
+    def __init__(
+        self,
+        wires=None,
+        shots=None,
+        check_clifford=True,
+        max_error=None,
+        tableau=True,
+        seed="global",
+        max_workers=None,
+    ) -> None:
+        super().__init__(wires=wires, shots=shots)
+
+        self._max_workers = max_workers
+        self._check_clifford = check_clifford
+
+        if max_error is None:
+            self._max_error = 0.0
+        else:
+            raise ValueError(
+                f"Currently this device doesn't support executing non-Clifford operations, got max_error={max_error}."
+            )
+
+        self._tableau = tableau
+
+        seed = np.random.randint(0, high=10000000) if seed == "global" else seed
+        self._rng = np.random.default_rng(seed)
```

Comment:
What if the user requests the JAX interface?

Reply:
We don't support the JAX interface with this PR. I can add it in the other PR where I have added the sampling support.